amp/WEB-INF/src/org/digijava/kernel/ampapi/endpoints/geocoding/GeoCoderEndpoint.java

```diff
@@ -0,0 +1,79 @@
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.digijava.kernel.ampapi.endpoints.errors.ApiError;
+import org.digijava.kernel.ampapi.endpoints.errors.ApiErrorResponse;
+import org.digijava.kernel.ampapi.endpoints.errors.ApiRuntimeException;
+import org.digijava.kernel.ampapi.endpoints.security.AuthRule;
+import org.digijava.kernel.ampapi.endpoints.util.ApiMethod;
+import org.digijava.kernel.entity.geocoding.GeoCoding;
+import org.digijava.kernel.geocoding.service.GeneralGeoCodingException;
+import org.digijava.kernel.geocoding.service.GeoCodingNotAvailableException;
+import org.digijava.kernel.geocoding.service.GeoCodingService;
+
+/**
+ * @author Octavian Ciubotaru
+ */
+@Path("geo-coder")
+@Api("geo-coder")
+public class GeoCoderEndpoint {
+
+    private final GeoCodingService service = new GeoCodingService();
+
+    @ApiOperation(value = "Geo code activities",
+            notes = "This operation will fail if geo coding is in use by another team member.")
+    @ApiResponses(@ApiResponse(code = HttpServletResponse.SC_NO_CONTENT, message = "success"))
+    @ApiMethod(id = "process", authTypes = AuthRule.IN_WORKSPACE)
+    @POST
+    @Path("process")
+    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
+    public Response process(
+            @ApiParam("Activity ids") Set<Long> activityIds) {
+        try {
+            service.processActivities(activityIds);
+        } catch (GeoCodingNotAvailableException e) {
+            ApiErrorResponse apiErrorResponse = ApiError.toError(
+                    GeoCoderEndpointErrors.GEO_CODING_NOT_AVAILABLE.withDetails(e.getTeamMember().toString()));
+            throw new ApiRuntimeException(Response.Status.BAD_REQUEST, apiErrorResponse);
+        } catch (GeneralGeoCodingException e) {
+            ApiErrorResponse apiErrorResponse = ApiError.toError(
+                    GeoCoderEndpointErrors.GEO_CODING_GENERAL_ERROR.withDetails(e.getMessage()));
+            throw new ApiRuntimeException(Response.Status.BAD_REQUEST, apiErrorResponse);
+        }
+        return Response.noContent().build();
+    }
+
+    @ApiOperation("Returns geo coding status")
+    @ApiResponses({
+            @ApiResponse(code = HttpServletResponse.SC_OK, message = "geo coding status", response = GeoCoding.class),
+            @ApiResponse(code = HttpServletResponse.SC_NOT_FOUND, message = "no geo coding in progress") })
+    @ApiMethod(id = "process", authTypes = AuthRule.IN_WORKSPACE)
+    @GET
+    @Path("status")
+    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
+    public GeoCoding getProcessingStatus() {
+        GeoCoding status = service.getProcessingStatus();
+        if (status == null) {
+            throw new WebApplicationException(
```

Comment:
We shouldn't throw an error if the no geocoding is in progress or is not available in current ws.
If the geocoding is not available for the current user, the status should be "NOT_AVAILABLE" with additional info about the user/ws
If no geocoding is present in the system, we could add a new status: "OPEN" or "AVAILABLE" for geocoding